TextureSwiftSupport/SpecBuilder.swift

```diff
@@ -288,3 +288,34 @@ public struct VSpacerLayout : _ASLayoutElementType {
     ]
   }
 }
+
+public struct StyledLayout<Content>: _ASLayoutElementType where Content : _ASLayoutElementType {
+  
+  let styling: (ASLayoutElementStyle) -> Void
+  let content: Content
+  
+  public init(styling: @escaping (ASLayoutElementStyle) -> Void, content: () -> Content) {
+    self.styling = styling
+    self.content = content()
+  }
+  
+  public func make() -> [ASLayoutElement] {
+    let elems = content.make()
+    elems.forEach { (elem) in
+      styling(elem.style)
+    }
+    return elems
+  }
+  
+}
+
+
+extension Array: _ASLayoutElementType where Element: _ASLayoutElementType {
```

Comment:
This is a great idea!

Reply:
This implementation has been implemented by https://github.com/muukii/TextureSwiftSupport/pull/3.